modules/apps/transfer/ibc_module.go

```diff
@@ -116,16 +122,16 @@ func (im IBCModule) OnChanOpenTry(
 		return "", err
 	}
 
-	if counterpartyVersion != types.Version {
-		return "", errorsmod.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: expected %s, got %s", types.Version, counterpartyVersion)
+	if !slices.Contains(types.SupportedVersions, counterpartyVersion) {
```

Comment:
Note: we could instead of erroring here, return our preferred version for ACK to optionally accept

Reply:
This should also be changed in ics20-1 only versions of ibc-go to truly take advantage of the change.

It will also help here: https://github.com/cosmos/ibc-go/issues/6171

Reply:
i think the comment from @chatton is relevant here, that relying on ics4wrapper to correctly return the transfer app version might not be the safest idea -- so if we want to return our preferred version here, we might want to consider adding the app version to the channel. wdyt?

Reply:
Can you link or summarize the comment from @chatton? Where is the ics4wrapper used when returning a string in the channel handshake?

Reply:
if we want to counterpropose with our preferred version for ACK to optionally accept, we will have to get our own version from the ICS4 wrappper `GetAppVersion` right? since we only get the counterparty version from the params.